build/patches/Experimental-user-scripts-support.patch

```diff
@@ -3087,12 +3142,12 @@ new file mode 100755
 +      return false;
 +    }
 +
-+    script->set_name(user_script_path.BaseName().value());
 +    script->set_match_origin_as_fallback(MatchOriginAsFallbackBehavior::kNever);
 +
 +    std::unique_ptr<UserScript::File> file(new UserScript::File());
 +    file->set_content(content);
-+    file->set_url(GURL(base::StrCat({"https://userscripts/file/", script->name(), ".js"})));
++    file->set_url(GURL(base::StrCat({"https://userscripts/file/",
```

Comment:
`name()` comes from here:
```
      } else if (GetDeclarationValue(line, kNameDeclaration, &value)) {
        script->set_name(value);
```

So it is the name as declared inside the source?

Reply:
And with this PR it becomes instead the basename? Will the extension be doubled? e.g. `script.user.js.js`

Reply:
uhm... maybe you're right. pause this pull I'll double check it.

Reply:
What is that URL used for anyways? I am trying to understand the pickle/unpickle procedure and I see that at some point it is overridden with a call to `set_url()`.

Reply:
> Was script->name() sanitized against attacks like using ../ and similar?

checked, it is not sanitized, also because there is no need for it.
the (dummy) url is generated one line below: `user_script_path.BaseName().value(), ".js"})));`

> And with this PR it becomes instead the basename?

no, it is the value taken directly from the js, as you indicated.

> Will the extension be doubled?

it might, but that's not a problem.

> What is that URL used for anyways?

it is used [here](https://github.com/bromite/bromite/blob/4d49af97d9ab8bbf9d2c8fc000721ca5c5a9db74/build/patches/Experimental-user-scripts-support.patch#L9615
)

Reply:
> That is a validation problem because invalid URL characters could be there. I do not think it is a security problem because these characters are not used for file paths and the URLs are not resolved.

see [here](https://github.com/bromite/bromite/blob/master/build/patches/Experimental-user-scripts-support.patch#L3090)

> Can you please refer in which comment I indicated this?

https://github.com/bromite/bromite/pull/1957#discussion_r846246631
```
      } else if (GetDeclarationValue(line, kNameDeclaration, &value)) {  <-----
        script->set_name(value);
```

> Do we need URLs at all? Why don't we use the hash of the script instead?

yes and of course it is possible, I try it so we also fix the possible double .js

Reply:
I do not understand that line you reference `script->set_name(user_script_path.BaseName().value());`; why is it being set twice? Who is using it with the other value before this line is called? Could this be avoided and `set_name()` only called once? Also, what is the name supposed to contain? the original base filename (`some script.user.js`) or the name (`Some script`)?

>> Can you please refer in which comment I indicated this?
> 
> https://github.com/bromite/bromite/pull/1957#discussion_r846246631

I do not understand why in this PR it is changed, could you please explain that?

Reply:
>> Do we need URLs at all? Why don't we use the hash of the script instead?
> 
> yes and of course it is possible, I try it so we also fix the possible double .js

Ok, perhaps it will simplify things a bit.

Reply:
> why is it being set twice? 

was an error

> Also, what is the name supposed to contain?

The name from script metadata (`Some script`)

> I do not understand why in this PR it is changed, could you please explain that?

simply because in the ui (java side) the name of the userscript was not shown but (incorrectly) the name of the file

Reply:
> Why don't we use the hash of the script instead?

fixed in https://github.com/bromite/bromite/commit/8b9bc0f0710adaf3da7520c9af73a12ccef23a45